app/src/main/kotlin/compiler/parser/grammar/ParserGrammar.kt

```diff
@@ -1,7 +1,109 @@
+            getProduction("EXPR2048", "\\{tIDENTIFIER}"),
+            getProduction("EXPR2048", "\\{tIDENTIFIER}$NLS\\{tLEFT_PAREN}$NLS\\{nCALL_ARGS}$NLS\\{tRIGHT_PAREN}"),
+            getProduction("EXPR2048", "\\{tMINUS}$NLS\\{EXPR2048}"),
+            getProduction("EXPR2048", "\\{tINCREMENT}$NLS\\{tIDENTIFIER}"),
+            getProduction("EXPR2048", "\\{tIDENTIFIER}$NLS\\{tINCREMENT}"),
+            getProduction("EXPR2048", "\\{tDECREMENT}$NLS\\{tIDENTIFIER}"),
+            getProduction("EXPR2048", "\\{tIDENTIFIER}$NLS\\{tDECREMENT}"),
+
+            getProduction("EXPR1024", "\\{nEXPR2048}"),
+            getProduction("EXPR1024", "\\{nEXPR2048}$NLS\\{tMULTIPLY}$NLS\\{nEXPR1024}"),
+            getProduction("EXPR1024", "\\{nEXPR2048}$NLS\\{tDIVIDE}$NLS\\{nEXPR1024}"),
+            getProduction("EXPR1024", "\\{nEXPR2048}$NLS\\{tMODULO}$NLS\\{nEXPR1024}"),
+
+            getProduction("EXPR512", "\\{nEXPR1024}"),
+            getProduction("EXPR512", "\\{nEXPR1024}$NLS\\{tPLUS}$NLS\\{nEXPR512}"),
+            getProduction("EXPR512", "\\{nEXPR1024}$NLS\\{tMINUS}$NLS\\{nEXPR512}"),
+
+            getProduction("EXPR256", "\\{nEXPR512}"),
+            getProduction("EXPR256", "\\{nEXPR512}$NLS\\{tSHIFT_LEFT}$NLS\\{nEXPR256}"),
+            getProduction("EXPR256", "\\{nEXPR512}$NLS\\{tSHIFT_RIGHT}$NLS\\{nEXPR256}"),
+
+            getProduction("EXPR128", "\\{nEXPR256}"),
+            getProduction("EXPR128", "\\{nEXPR256}$NLS\\{tBIT_AND}$NLS\\{nEXPR128}"),
+
+            getProduction("EXPR64", "\\{nEXPR128}"),
+            getProduction("EXPR64", "\\{nEXPR128}$NLS\\{tBIT_XOR}$NLS\\{nEXPR64}"),
+
+            getProduction("EXPR32", "\\{nEXPR64}"),
+            getProduction("EXPR32", "\\{nEXPR64}$NLS\\{tBIT_OR}$NLS\\{nEXPR32}"),
+
+            getProduction("EXPR16", "\\{nEXPR32}"),
+            getProduction("EXPR16", "\\{nEXPR32}$NLS\\{tEQUAL}$NLS\\{nEXPR16}"),
+            getProduction("EXPR16", "\\{nEXPR32}$NLS\\{tNOT_EQUAL}$NLS\\{nEXPR16}"),
+            getProduction("EXPR16", "\\{nEXPR32}$NLS\\{tLESS_THAN}$NLS\\{nEXPR16}"),
+            getProduction("EXPR16", "\\{nEXPR32}$NLS\\{tLESS_THAN_EQ}$NLS\\{nEXPR16}"),
+            getProduction("EXPR16", "\\{nEXPR32}$NLS\\{tGREATER_THAN}$NLS\\{nEXPR16}"),
+            getProduction("EXPR16", "\\{nEXPR32}$NLS\\{tGREATER_THAN_EQ}$NLS\\{nEXPR16}"),
+
+            getProduction("EXPR8", "\\{nEXPR16}"),
+            getProduction("EXPR8", "\\{nEXPR16}$NLS\\{tXOR}$NLS\\{nEXPR8}"),
+            getProduction("EXPR8", "\\{nEXPR16}$NLS\\{tIFF}$NLS\\{nEXPR8}"),
+
+            getProduction("EXPR4", "\\{nEXPR8}"),
+            getProduction("EXPR4", "\\{nEXPR8}$NLS\\{tAND}$NLS\\{nEXPR4}"),
+
+            getProduction("EXPR2", "\\{nEXPR4}"),
+            getProduction("EXPR2", "\\{nEXPR4}$NLS\\{tOR}$NLS\\{nEXPR2}"),
+
+            getProduction("EXPR", "\\{nEXPR2}"),
+            getProduction("EXPR", "\\{nEXPR2}$NLS\\{tQUESTION_MARK}$NLS\\{nEXPR2}$NLS\\{tCOLON}$NLS\\{nEXPR}"),
```

Comment:
I've been trying to run this through the Parser from existing PRs and I keep getting errors. Let's say we want to deduce `EXPR` from sequence `EXPR2, NEWLINE, ...`. After reading `EXPR2` we can either shift (consume `NEWLINE` and step in the dfa) or reduce with production `EXPR -> EXPR2` as `NEWLINE` is part of `folllow[EXPR]`. Am I understanding this wrong so is the Parser or is this grammar not LL(1)?

Reply:
You're right, the grammar is not LL(1). I managed to solve the problem with EXPR and NEWLINE you have found (although it's an ugly solution), but another one came out. We'll try to fix it on Sunday.

Reply:
Yeah, it was just an example. The parser returned many similar errors, mostly based on the newline symbol.